### PR TITLE
Fixes incorrect "more" calculation for attack speed #380

### PR DESCRIPTION
**Pull request by jrobinson3k1**
I needed to convert the percentage values from whole numbers to their decimal representation in order for the calculation to be correct.

### CODE REVIEW

**Comment by EmmittJ** [comment]
I think I stated in the issue something wrong, sorry. I believe the less attack speed here is wrong since you need to multiple the product with the more attack speed product.

**Comment by jrobinson3k1** [comment]
@EmmittJ should be good now

**Comment by EmmittJ** [comment]
What happens when you have an odd amount of less attack speed? I believe
you will get on overall negative attack speed?

**Comment by jrobinson3k1** [comment]
Yeah, I think there's a deeper problem with how "more" and "less" values are stored in the attrs map. I'm not sure where those values are obtained in source, but I assume the collective value is just added together for the "#% less Attack Speed" and "#% more Attack Speed" value?

**Comment by jrobinson3k1** [comment]
The "more" and "less" values are calculated incorrect for every value that I'm seeing, so maybe it's best to discard the PR and create a new issue to address the global problem and fix it all in one fell swoop.

**Comment by EmmittJ** [comment]
Yeah, I can correct it. The two values need to be calculated separately
(products) then the product of those is the final result.

File: WPFSKillTree/SkillTreeFiles/Compute.cs [hunk 1]
```diff
@@ -624,23 +624,20 @@ public void AttackSpeed(AttackSkill skill, AttributeSet attrs)
...
                         APS = IncreaseValueByPercentage(APS, incAS);
 
-                    float moreAS = 0;
+                    float moreAS = 1;
                     if (attrs.ContainsKey("#% more Attack Speed"))
-                        moreAS += attrs["#% more Attack Speed"][0];
+                        moreAS *= attrs["#% more Attack Speed"][0] / 100 + 1;
                     if (attrs.ContainsKey("#% less Attack Speed"))
-                        moreAS -= attrs["#% less Attack Speed"][0];
+                        moreAS /= attrs["#% less Attack Speed"][0] / 100 + 1;
```
Comment (MLanghof) [inline WPFSKillTree/SkillTreeFiles/Compute.cs]: I'm pretty sure this should be `moreAS *= 1 - attrs[...]/100;`.
100% less attack speed means you never attack, not that you attack half as often.

**Comment by EmmittJ** [comment]
Yep

File: WPFSKillTree/SkillTreeFiles/Compute.cs [hunk 2]
```diff
@@ -624,23 +624,20 @@ public void AttackSpeed(AttackSkill skill, AttributeSet attrs)
...
                         if (m.Success)
-                        {
                             // XXX: Not sure there are any mods with WeaponType here (Melee string in mod is DamageForm now, maybe Unarmed should be form as well).
-                            if (Weapon.Types.ContainsKey(m.Groups[2].Value) && Nature.Is(Weapon.Types[m.Groups[2].Value]))
-                                moreAS += m.Groups[1].Value == "more" ? attr.Value[0] : -attr.Value[0];
-                            else if (DamageNature.Forms.ContainsKey(m.Groups[2].Value) && Nature.Is(DamageNature.Forms[m.Groups[2].Value]))
-                                moreAS += m.Groups[1].Value == "more" ? attr.Value[0] : -attr.Value[0];
-                        }
+                            if (Weapon.Types.ContainsKey(m.Groups[2].Value) && Nature.Is(Weapon.Types[m.Groups[2].Value]) || DamageNature.Forms.ContainsKey(m.Groups[2].Value) && Nature.Is(DamageNature.Forms[m.Groups[2].Value]))
+                                moreAS *= (m.Groups[1].Value == "more" ? attr.Value[0] : -attr.Value[0]) / 100 + 1;
```
Comment (l0g0sys) [inline WPFSKillTree/SkillTreeFiles/Compute.cs]: Same here. There is condition for RegExp subgroup "more" to treat mod as "more AS" otherise as "less AS"

**Pull request closed without merging**